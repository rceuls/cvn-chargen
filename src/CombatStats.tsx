import { ICharacter } from './services/character.generator';
import { getModifier, getRndInteger } from './services/util';

const CombatStats = ({ character }: { character: ICharacter }) => {
  const hp =
    getRndInteger(3, character.charClass.startingHP) +
    getModifier(character.attributes.con);

  return (
    <>
      <div className="row">
        <div className="col-sm">
          <span className="font-weight-bold">
            Base Attack Bonus (melee, ranged)
          </span>
          <span className="ml-2">
            +0 ({getModifier(character.attributes.str)}/
            {getModifier(character.attributes.dex)})
          </span>
          <span className="ml-2 font-weight-bold">Armor Class</span>
          <span className="ml-2">
            {10 +
              character.charClass.equipment.acMod +
              getModifier(character.attributes.dex)}
          </span>
          <span className="ml-2 font-weight-bold">Hit Points</span>
          <span className="ml-2">{hp > 0 ? hp : 1}</span>
        </div>
      </div>
    </>
  );
};

export default CombatStats;
