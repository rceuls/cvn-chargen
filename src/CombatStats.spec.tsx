import renderer from 'react-test-renderer';
import CombatStats from './CombatStats';

it('renders correctly', () => {
  const tree = renderer
    .create(
      <CombatStats
        character={{
          armorClass: 12,
          maximumHP: 13,
          attributeScores: [8, 18, 10, 10, 10],
        }}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
