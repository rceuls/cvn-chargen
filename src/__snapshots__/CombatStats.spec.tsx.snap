// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <div
    className="row"
  >
    <div
      className="col-sm"
    >
      <span
        className="font-weight-bold"
      >
        THAC0 (melee, ranged)
      </span>
      <span
        className="ml-2"
      >
        19 (
        -1
        ,
        +3
        )
      </span>
    </div>
  </div>,
  <div
    className="row"
  >
    <div
      className="col-sm"
    >
      <span
        className="font-weight-bold"
      >
        Armor Class
      </span>
      <span
        className="ml-2"
      >
        12
      </span>
      <span
        className="ml-2 font-weight-bold"
      >
        Maximum Hit Points
      </span>
      <span
        className="ml-2"
      >
        13
      </span>
    </div>
  </div>,
  <hr />,
]
`;
